VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.tsx

```diff
@@ -1,31 +1,56 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { useSelector } from 'react-redux'
 
 import { StackScreenProps } from '@react-navigation/stack'
 
 import { Button } from '@department-of-veterans-affairs/mobile-component-library'
 
+import { useDownloadLetter, useLetterBeneficiaryData } from 'api/letters'
+import { LetterTypeConstants, LettersDownloadParams } from 'api/types'
 import { AlertBox, BasicError, Box, FeatureLandingTemplate, LoadingComponent, TextArea, TextView } from 'components'
 import { NAMESPACE } from 'constants/namespaces'
 import { BenefitsStackParamList } from 'screens/BenefitsScreen/BenefitsStackScreens'
-import { RootState } from 'store'
-import { LetterTypeConstants } from 'store/api/types'
-import { LettersState, downloadLetter } from 'store/slices'
 import { testIdProps } from 'utils/accessibility'
 import { generateTestID } from 'utils/common'
-import { useAppDispatch, useTheme } from 'utils/hooks'
+import { useTheme } from 'utils/hooks'
 
 type GenericLetterProps = StackScreenProps<BenefitsStackParamList, 'GenericLetter'>
 
 function GenericLetter({ navigation, route }: GenericLetterProps) {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
-  const dispatch = useAppDispatch()
   const { header, description, letterType, descriptionA11yLabel } = route.params
-  const { downloading, letterDownloadError } = useSelector<RootState, LettersState>((state) => state.letters)
+  const [downloadLetter, setDownloadLetter] = useState(false)
+  const { data: letterBeneficiaryData } = useLetterBeneficiaryData()
+  const lettersOptions: LettersDownloadParams = {
+    militaryService: false,
+    serviceConnectedDisabilities: false,
+    serviceConnectedEvaluation: false,
+    nonServiceConnectedPension: letterBeneficiaryData?.benefitInformation.hasNonServiceConnectedPension || false,
+    monthlyAward: false,
+    unemployable: letterBeneficiaryData?.benefitInformation.hasIndividualUnemployabilityGranted || false,
+    specialMonthlyCompensation: letterBeneficiaryData?.benefitInformation.hasSpecialMonthlyCompensation || false,
+    adaptedHousing: letterBeneficiaryData?.benefitInformation.hasAdaptedHousing || false,
+    chapter35Eligibility: false,
+    deathResultOfDisability: letterBeneficiaryData?.benefitInformation.hasDeathResultOfDisability || false,
+    survivorsAward:
+      letterBeneficiaryData?.benefitInformation.hasSurvivorsIndemnityCompensationAward ||
+      letterBeneficiaryData?.benefitInformation.hasSurvivorsPensionAward ||
+      false,
+  }
```

Comment:
Instead of declaring the letter options in multiple places, you can set this as a const directly in the `useDownloadLetter` query, and just grab the letter beneficiary data from the query client using `getQueryData`. 

If the `useDownloadLetter` query is being called, the letter beneficiary data should already be cached. This will definitely be the case with the home screen personalization changes. But if you want to be safe, you could always check if cached data for it exists and just fetch the query with the query client if it doesn't

Reply:
If the default letter options were stored in the hook, there would be a single source of truth for letter options, which would be beneficial in regard to maintenance and accuracy. If more options need to be added in the future, only one place would need to be updated instead of multiple places, which would prevent the default letter options from being out of sync across multiple files. I'm assuming that's why the slice had the default options stored there also. 

Since it's currently only used in two different screens, it's probably not as big of a deal right now, but it potentially could in the future. I'm fine with either path right now though, it can always be revisited in the future if letter options grow, just wanted to call that out.

Reply:
opting to leave it as is right now.